File: core/core-designsystem/src/androidMain/kotlin/dev/franzueto/fluxit/core/designsystem/components/FluxItListItem.kt
package dev.franzueto.fluxit.core.designsystem.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.style.TextDecoration
import androidx.compose.ui.unit.dp
import dev.franzueto.fluxit.core.designsystem.tokens.FluxItColors
import dev.franzueto.fluxit.core.designsystem.tokens.FluxItSpacing
import dev.franzueto.fluxit.core.designsystem.tokens.FluxItTypography

// Variant 1 — Dashboard list-item: 56dp tinted icon container, title+subtitle,
// optional trash + chevron trailing slot.
@Composable
@Suppress("ktlint:standard:function-naming")
public fun FluxItDashboardListItem(
    icon: ImageVector,
    iconTint: Color,
    title: String,
    subtitle: String? = null,
    onClick: () -> Unit = {},
    trashIcon: ImageVector? = null,
    onDelete: (() -> Unit)? = null,
    chevronIcon: ImageVector? = null,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier =
            modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(16.dp))
                .background(FluxItColors.surfaceCard)
                .clickable(onClick = onClick)
                .padding(horizontal = FluxItSpacing.itemPaddingX, vertical = FluxItSpacing.itemPaddingY),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Box(
            modifier =
                Modifier
                    .size(56.dp)
                    .clip(RoundedCornerShape(14.dp))
                    .background(iconTint.copy(alpha = 0.2f)),
            contentAlignment = Alignment.Center,
        ) {
            Icon(imageVector = icon, contentDescription = null, tint = iconTint)
        }
        Column(modifier = Modifier.weight(1f)) {
            Text(text = title, style = FluxItTypography.titleMd, color = FluxItColors.textPrimary)
            if (subtitle != null) {
                Text(text = subtitle, style = FluxItTypography.labelSm, color = FluxItColors.textMuted)
            }
        }
        if (trashIcon != null && onDelete != null) {
            IconButton(onClick = onDelete) {
                Icon(imageVector = trashIcon, contentDescription = "Delete", tint = FluxItColors.textMuted)
            }
        }
        if (chevronIcon != null) {
            Icon(imageVector = chevronIcon, contentDescription = null, tint = FluxItColors.textMuted)
        }
    }
}

// Variant 2 — Detail to-buy item: radio leading, chevron trailing.
@Composable
@Suppress("ktlint:standard:function-naming")
public fun FluxItToBuyListItem(
    title: String,
    onToggle: () -> Unit,
    trailingIcon: ImageVector? = null,
    onClick: () -> Unit = {},
    modifier: Modifier = Modifier,
) {
    Row(
        modifier =
            modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(12.dp))
                .background(FluxItColors.surfaceCardMuted)
                .clickable(onClick = onClick)
                .padding(horizontal = FluxItSpacing.itemPaddingX, vertical = FluxItSpacing.itemPaddingY),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Box(
            modifier =
                Modifier
                    .size(24.dp)
                    .clip(CircleShape)
                    .border(width = 2.dp, color = FluxItColors.textMuted, shape = CircleShape)
                    .clickable(onClick = onToggle),
        )
        Text(
            text = title,
            style = FluxItTypography.bodyMd,
            color = FluxItColors.textPrimary,
            modifier = Modifier.weight(1f),
        )
        if (trailingIcon != null) {
            Icon(imageVector = trailingIcon, contentDescription = null, tint = FluxItColors.textMuted)
        }
    }
}

// Variant 3 — Detail completed item: filled check leading, strikethrough title,
// trash trailing.
@Composable
@Suppress("ktlint:standard:function-naming")
public fun FluxItCompletedListItem(
    title: String,
    onToggle: () -> Unit,
    checkIcon: ImageVector,
    trashIcon: ImageVector,
    onDelete: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier =
            modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(12.dp))
                .background(FluxItColors.surfaceCardMuted)
                .padding(horizontal = FluxItSpacing.itemPaddingX, vertical = FluxItSpacing.itemPaddingY),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        IconButton(onClick = onToggle, modifier = Modifier.size(24.dp)) {
            Icon(imageVector = checkIcon, contentDescription = "Completed", tint = FluxItColors.primaryBlue)
        }
        Text(
            text = title,
            style = FluxItTypography.bodyMd.copy(textDecoration = TextDecoration.LineThrough),
            color = FluxItColors.textMuted,
            modifier = Modifier.weight(1f),
        )
        IconButton(onClick = onDelete, modifier = Modifier.size(24.dp)) {
            Icon(imageVector = trashIcon, contentDescription = "Delete", tint = FluxItColors.textMuted)
        }
    }
}

File: ios-app/Sources/DesignSystem/Components/FluxItListItem.swift
import SwiftUI

// Variant 1 — Dashboard list-item: 56pt tinted icon container, title+subtitle,
// optional trash + chevron trailing slots.
public struct FluxItDashboardListItem: View {
    private let icon: Image
    private let iconTint: Color
    private let title: String
    private let subtitle: String?
    private let onTap: () -> Void
    private let trashIcon: Image?
    private let onDelete: (() -> Void)?
    private let chevronIcon: Image?

    public init(
        icon: Image,
        iconTint: Color,
        title: String,
        subtitle: String? = nil,
        onTap: @escaping () -> Void = {},
        trashIcon: Image? = nil,
        onDelete: (() -> Void)? = nil,
        chevronIcon: Image? = nil
    ) {
        self.icon = icon
        self.iconTint = iconTint
        self.title = title
        self.subtitle = subtitle
        self.onTap = onTap
        self.trashIcon = trashIcon
        self.onDelete = onDelete
        self.chevronIcon = chevronIcon
    }

    public var body: some View {
        Button(action: onTap) {
            HStack(spacing: 12) {
                ZStack {
                    RoundedRectangle(cornerRadius: 14, style: .continuous)
                        .fill(iconTint.opacity(0.2))
                    icon.foregroundStyle(iconTint)
                }
                .frame(width: 56, height: 56)
                VStack(alignment: .leading, spacing: 2) {
                    Text(title)
                        .font(.system(size: 18, weight: .semibold))
                        .foregroundStyle(FluxItTokens.Colors.textPrimary)
                    if let subtitle {
                        Text(subtitle)
                            .font(.system(size: 14))
                            .foregroundStyle(FluxItTokens.Colors.textMuted)
                    }
                }
                .frame(maxWidth: .infinity, alignment: .leading)
                if let trashIcon, let onDelete {
                    Button(action: onDelete) {
                        trashIcon.foregroundStyle(FluxItTokens.Colors.textMuted)
                    }
                    .buttonStyle(.plain)
                    .accessibilityLabel("Delete")
                }
                if let chevronIcon {
                    chevronIcon.foregroundStyle(FluxItTokens.Colors.textMuted)
                }
            }
            .padding(.horizontal, FluxItTokens.Spacing.itemPaddingX)
            .padding(.vertical, FluxItTokens.Spacing.itemPaddingY)
            .background(FluxItTokens.Colors.surfaceCard)
            .clipShape(RoundedRectangle(cornerRadius: 16, style: .continuous))
        }
        .buttonStyle(.plain)
    }
}

// Variant 2 — Detail to-buy item: hollow radio leading, chevron trailing.
public struct FluxItToBuyListItem: View {
    private let title: String
    private let onToggle: () -> Void
    private let onTap: () -> Void
    private let chevronIcon: Image?

    public init(
        title: String,
        onToggle: @escaping () -> Void,
        onTap: @escaping () -> Void = {},
        chevronIcon: Image? = nil
    ) {
        self.title = title
        self.onToggle = onToggle
        self.onTap = onTap
        self.chevronIcon = chevronIcon
    }

    public var body: some View {
        Button(action: onTap) {
            HStack(spacing: 12) {
                Button(action: onToggle) {
                    Circle()
                        .strokeBorder(FluxItTokens.Colors.textMuted, lineWidth: 2)
                        .frame(width: 24, height: 24)
                        .frame(width: 44, height: 44) // expand tap area without changing the visual
                        .contentShape(Rectangle())
                }
                .buttonStyle(.plain)
                .accessibilityLabel("Mark as completed")
                Text(title)
                    .font(.system(size: 16))
                    .foregroundStyle(FluxItTokens.Colors.textPrimary)
                    .frame(maxWidth: .infinity, alignment: .leading)
                if let chevronIcon {
                    chevronIcon.foregroundStyle(FluxItTokens.Colors.textMuted)
                }
            }
            .padding(.horizontal, FluxItTokens.Spacing.itemPaddingX)
            .padding(.vertical, FluxItTokens.Spacing.itemPaddingY)
            .background(FluxItTokens.Colors.surfaceCardMuted)
            .clipShape(RoundedRectangle(cornerRadius: 12, style: .continuous))
        }
        .buttonStyle(.plain)
    }
}

// Variant 3 — Detail completed item: filled check leading, strikethrough title,
// trash trailing.
public struct FluxItCompletedListItem: View {
    private let title: String
    private let onToggle: () -> Void
    private let checkIcon: Image
    private let trashIcon: Image
    private let onDelete: () -> Void

    public init(
        title: String,
        onToggle: @escaping () -> Void,
        checkIcon: Image,
        trashIcon: Image,
        onDelete: @escaping () -> Void
    ) {
        self.title = title
        self.onToggle = onToggle
        self.checkIcon = checkIcon
        self.trashIcon = trashIcon
        self.onDelete = onDelete
    }

    public var body: some View {
        HStack(spacing: 12) {
            Button(action: onToggle) {
                checkIcon
                    .foregroundStyle(FluxItTokens.Colors.primaryBlue)
                    .frame(width: 44, height: 44)
                    .contentShape(Rectangle())
            }
            .buttonStyle(.plain)
            .accessibilityLabel("Completed")
            Text(title)
                .font(.system(size: 16))
                .strikethrough()
                .foregroundStyle(FluxItTokens.Colors.textMuted)
                .frame(maxWidth: .infinity, alignment: .leading)
            Button(action: onDelete) {
                trashIcon
                    .foregroundStyle(FluxItTokens.Colors.textMuted)
                    .frame(width: 44, height: 44)
                    .contentShape(Rectangle())
            }
            .buttonStyle(.plain)
            .accessibilityLabel("Delete")
        }
        .padding(.horizontal, FluxItTokens.Spacing.itemPaddingX)
        .padding(.vertical, FluxItTokens.Spacing.itemPaddingY)
        .background(FluxItTokens.Colors.surfaceCardMuted)
        .clipShape(RoundedRectangle(cornerRadius: 12, style: .continuous))
    }
}
